src/modules/md/process.cpp

```diff
@@ -141,25 +148,26 @@ bool MDModule::process(Dissolve &dissolve, ProcessPool &procPool)
         massSum += m;
     }
 
-    // Remove any velocity shift, and re-zero velocities on fixed atoms
-    vCom /= massSum;
-    std::transform(velocities.begin(), velocities.end(), velocities.begin(), [vCom](auto vel) { return vel - vCom; });
-    for (auto &&[v, iFree] : zip(velocities, free))
-        if (!iFree)
-            v.zero();
+    // Finalise initial velocities (unless considering intramolecular forces only)
+    if (!intramolecularForcesOnly_)
+    {
+        // Remove any velocity shift, and re-zero velocities on fixed atoms
+        vCom /= massSum;
+        std::transform(velocities.begin(), velocities.end(), velocities.begin(), [vCom](auto vel) { return vel - vCom; });
+        for (auto &&[v, iFree] : zip(velocities, free))
+            if (!iFree)
+                v.zero();
 
-    // Calculate instantaneous temperature
-    // J = kg m2 s-2  -->   10 J = g Ang2 ps-2
-    // If ke is in units of [g mol-1 Angstroms2 ps-2] then must use kb in units of 10 J mol-1 K-1 (= 0.8314462)
-    const auto kb = 0.8314462;
-    ke = 0.0;
-    for (auto &&[m, v] : zip(mass, velocities))
-        ke += 0.5 * m * v.dp(v);
-    tInstant = ke * 2.0 / (3.0 * atoms.size() * kb);
+        // Calculate instantaneous temperature
+        ke = 0.0;
+        for (auto &&[m, v] : zip(mass, velocities))
+            ke += 0.5 * m * v.dp(v);
```

Comment:
```suggestion
        std::transform_reduce(ParallelPolicies::par_unseq, mass.begin(), mass.end(), velocities.begin(), 0.0, std::plus<double>(), [](double m, double v){return 0.5 * m * v.dp(v);});
```

Even if the arrays are small, this still might manage to benefit from an `unseq` policy.
Depending on the